tests/docs/multi_dut_asic_testing_support.md

```diff
@@ -0,0 +1,499 @@
+* all DUTs
+* specific type of DUTs
+* specific DUT
+* all ASICs
+* specific ASIC
+* combinations of the above scenarios
+
+Meanwhile, we need to ensure backward compatibility for scenarios including:
+* single DUT, single ASIC
+* Multiple single-ASIC DUTs
+
+The test scripts should be able to use interfaces like below to perform operations on DUTs and ASICs
+```python
+duthost.foo()
+duthost.foo(asic_index=1)
+duthost.asics[1].foo()      # Equivalent to the line above
+duthost.foo(asic_inex="all")
+
+[asic.foo() for asic in duthost.asics]
+
+duthosts.foo()
+duthosts.foo(asic_index=1)
+duthosts.foo(asic_index="all")
+
+[node.foo() for node in duthosts]
+
+duthosts.nodes.foo()                 # Equivalent to: duthosts.foo()
+duthosts.nodes.foo(asic_index=1)     # Equivalent to: duthosts.foo(asic_index=1)
+duthosts.nodes.foo(asic_index="all") # Equivalent to: duthosts.foo(asic_index="all")
+
+[node.foo() for node in duthosts.nodes]
+
+duthosts[1].foo()
+duthosts[1].foo(asic_index=1)
+duthosts[1].asics[1].foo()        # Equivalent to the line above
+duthosts[1].foo(asic_index="all")
+
+duthosts["node1"].foo()
+duthosts["node1"].foo(asic_index=1)
+duthosts["node1"].asics[1].foo()        # Equivalent to the line above
+duthosts["node1"].foo(asic_index="all")
+
+duthosts.nodes[1].foo()                 # Equivalent to: duthosts[1].foo()
+duthosts.nodes[1].foo(asic_index=1)     # Equivalent to: duthosts[1].foo(asic_index=1)
+duthosts.nodes[1].asics[1].foo()        # Equivalent to: duthosts[1].asics[1].foo()
+duthosts.nodes[1].foo(asic_index="all") # Equivalent to: duthosts[1].foo(asic_index="all")
+
+duthosts.nodes["node1"].foo()                 # Equivalent to: duthosts["node1"].foo()
+duthosts.nodes["node1"].foo(asic_index=1)     # Equivalent to: duthosts["node1"].foo(asic_index=1)
+duthosts.nodes["node1"].asics[1].foo()        # Equivalent to: duthosts["node1"].asics[1].foo()
+duthosts.nodes["node1"].foo(asic_index="all") # Equivalent to: duthosts["node1"].foo(asic_index="all")
+
+duthosts.frontend_nodes.foo()
+duthosts.frontend_nodes.foo(asic_index=1)
+duthosts.frontend_nodes.foo(asic_index="all")
+
+[node.foo() for node in duthosts.frontend_nodes]
+
+duthosts.frontend_nodes[1].foo()
+duthosts.frontend_nodes[1].foo(asic_index=1)
+duthosts.frontend_nodes[1].asics[1].foo()  # Equivalent to the line above
+duthosts.frontend_nodes[1].foo(asic_index="all")
+
+duthosts.frontend_nodes["node1"].foo()
+duthosts.frontend_nodes["node1"].foo(asic_index=1)
+duthosts.frontend_nodes["node1"].asics[1].foo()  # Equivalent to the line above
+duthosts.frontend_nodes["node1"].foo(asic_index="all")
+
+duthosts.supervisor_nodes.bar()
+duthosts.supervisor_nodes[0].bar()
+duthosts.supervisor_nodes["supervisor0"].bar()
+```
+
+In the above examples:
+* `duthost.asics` should be a list of objects. Each object represents an ASIC in the DUT. `duthost.asics` represents all the ASIC instances in the DUT.
```

Comment:
Some tests are applicable only for the frontend asics, and there could be some test cases that apply only for backend. We should also have:

duthost.frontend_asics
duthost.foo(asic_index="frontend_asics")

Reply:
Sure, we can enhance it this way.